scipy/signal/tests/test_filter_design.py

```diff
@@ -3734,3 +3734,73 @@ def test_w_or_N_types(self):
             w_out, gd = group_delay((1, 1), w)
             assert_array_almost_equal(w_out, [8])
             assert_array_almost_equal(gd, [0])
+
+
+class TestGammatone(object):
+    # Test erroneus input cases.
+    def test_invalid_input(self):
+        # Cutoff frequency is <= 0 or >= fs / 2.
+        fs = 16000
+        assert_raises(ValueError, gammatone, -fs, 'iir', fs=fs)
+        assert_raises(ValueError, gammatone, 0, 'fir', fs=fs)
+        assert_raises(ValueError, gammatone, fs / 2, 'iir', fs=fs)
+        assert_raises(ValueError, gammatone, fs, 'fir', fs=fs)
+
+        # Filter type is not fir or iir
+        assert_raises(ValueError, gammatone, 440, ftype='fie', fs=16000)
```

Comment:
It's clearer to do something like:
```
with pytest.raises(ValueError, match='ftype must be'):
    gammatone(440, ftype='fie', fs=16000)
```
Most of our tests don't have this because historically we couldn't make use of `pytest.raises`, but now we can, so newer code can benefit from it.

Reply:
Fixed. I switched the remaining `assert_raises` tests to `pytest.raises` and also switched the `assert_approx_equal` tests to use `pytest.approx`.